### DISCUSSION TITLE
Bulk add labels to labelling service

### DISCUSSION THREAD

**Discussion by jazmichaelking** [bluesky-social/atproto#2335] [General Discussion]
I'm hoping to add a defined set of labels to displace/replace/augment TBD the existing Bluesky labels

is there a way to add and define a label set (eg https://github.com/iftas-org/resources/tree/main/LABELS )

**Comment by bnewbold** [comment]
Great question!

The current setup is that there are a small number of "app-defined" labels, but otherwise each labeler defines it's own set of labels. The main distinction with "app-defined" labels is that they can be "self-labels" applied by accounts on their own content; there is no "label service" involved in that context, so there need to be app-level definitions.

You could definitely take the IFTAS labels/categories and turn them in to labels for an atproto/Bluesky labeling service.

An example service doing something similar is `aegis.blue`. Here is there label definition file, as JSON:

https://api.bsky.app/xrpc/com.atproto.repo.getRecord?repo=aegis.blue&collection=app.bsky.labeler.service&rkey=self

Label services currently edit files like thus via a big text field in Ozone.

One consideration that has come up in the past is whether it is appropriate to have labels for content which should generally just be taken down (for example, NCII). There is a trade-off between transparency and communicating "reasons" (in particular for aggregate reporting and communication between labeling services), versus negative externalities or drawing attention to some categories of content. There may be mechanisms to control access to some labels or labelers in the future, but currently all labels are public by default. Personally I think having the label category defined does little harm and could make it easier to report specific categories of content (in a future iteration of the reporting API), but there has been some misunderstanding the community around label definitions and policies, so we have leaned towards not defining labels for categories that we just do takedowns for.

It would potentially be quite useful to have the IFTAS labels defined publicly in JSON, possibly with translations (i18n) of the definitions, even if IFTAS does not run a labeler service, so that other services can re-use these labels and definitions in a consistent way.

Happy to discuss more!

**Reply by bnewbold** [reply]
A minor thing to note: if copying the linked labels to atproto, the currently enforced syntax for labels is lower-case, so `account-takeover` not `Account-Takeover`.

**Reply by jonpincus** [reply]
The editing in the UI is somewhat tedious -- there's not even a way to copy values.  Is there a way to upload the JSON definition, or even provide it in a file on the server?

**Reply by jazmichaelking** [reply]
Translations are on the to-do list for sure. Noted on the lower case. Need to think through the definitions for takedowns, we use labels for reporting _and_ reasons, although not all labels get reasons sent to an infringing user (ie simple takedown). We could work around the NCII/CSAM/TVEC etc for now, come back to those later. We're trying to separate policy from a common vocabulary, in other words we want people to be thinking about _all_ these labels, and their policies for them, without directing the policy (although of course we encourage certain policies...)

Will come back around to getting this in JSON, will share asap. And to be clear, most of these definitions are DTSP, not IFTAS, we're just working with pushing that forward as a common vocab.

**Reply by bnewbold** [reply]
@jonpincus I opened an issue to track ability to upload full JSON docs more easily: https://github.com/bluesky-social/ozone/issues/60

**Reply by jonpincus** [reply]
Thanks for this @bnewbold
